### PR TITLE
fix: wire glue layer for document delete

### PR DESCRIPTION
**Pull request by pleacu**
Fixes: #1555

Addressed previous pull issues.

### CODE REVIEW

File: ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx [hunk 1]
```diff
@@ -186,6 +187,7 @@ export function useAtlasmap({
   sources: IAtlasmapDocument[];
   targets: IAtlasmapDocument[];
   mappings: IMappings[];
+  deleteAtlasFile: (fileName: any, isSource: any) => void;
```
Comment (riccardo-forina) [inline ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: String and bool?

Comment (pleacu) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: The UI side returns a Mutable string | undefined which doesn't match string.  I didn't want to proliferate that mutable type into the provider side.

Comment (riccardo-forina) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: I’d still explicitly set the types and either wrap the calling functions in an if to check for undefined or use the ! operator to suppress the warning

Comment (pleacu) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: hmm - how does the ! operator work exactly?? :)

Comment (riccardo-forina) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Comment (pleacu) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: Added new PR

Comment (riccardo-forina) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: Erh... I was suggesting the other way round, checking that current is not null and making the type required in the delete function

Comment (pleacu) [reply ui-react/packages/atlasmap-provider/src/AtlasmapProvider.tsx]: Doh!  I'll give it another shot...

**Pull request merged**